### ISSUE TITLE
[Content Fragment] ElementDataSourceServlet does not get correct contentType for ContentElement on AEM 6.3

### ISSUE THREAD

**Issue by sanderlenaerts** [adobe/aem-core-wcm-components#849]
## Bug Report
**Current Behavior**
ElementDataSourceServlet does not work for getting the elementNames of a content fragment on AEM 6.3.3.6. 

/mnt/overlay/shared/granite/ui/content/foundation/form/content-fragment/picker/items/form/items/column/items/elementNames.html?fragmentPath=%2Fcontent%2Fdam%2Fcommon-content%2Ftestfragment&displayMode=singleText&_=1574149655383

The contentType  received from `ContentElement's getValue().getContentType() ` returns null for text items. 

**Expected behavior/code**
Return text/plain

**Environment**
- AEM 6.3.3.6
- Core Components 2.7.0

**Possible Solution**
After setting the contentType check if contentType is null and use element.getContentType to instantiate it correctly:

```
String contentType = element.getValue().getContentType();
               if (contentType == null) {
                    contentType = element.getContentType();
                }
```

**Comment by bpauli** [comment]
@sanderlenaerts, Thank you for providing this PR. I'm currently try to review your changes but I'm unable to reproduce the issue on my local AEM 6.3 instance. Can you please share some informations how you set up the testfragment and which content model you are using? Thanks!

**Comment by sanderlenaerts** [comment]
@bpauli I have created a new Content Fragment Model through the UI containing a single text field. 

Based on that CFM I have created the testfragment:
[testfragment-corecomponents.zip](https://github.com/adobe/aem-core-wcm-components/files/3865390/testfragment-corecomponents.zip)

**Comment by richardhand** [comment]
The code for this issue is merged to master and will be available in the upcoming [2.8.0 release](https://github.com/adobe/aem-core-wcm-components/milestone/18).
